src/Uno.Extensions.Navigation.UI/FrameworkElementExtensions.cs

```diff
@@ -1,10 +1,66 @@
-﻿using Uno.Extensions.Navigation.Regions;
-
+﻿using Uno.Extensions.Hosting;
+using Uno.Extensions.Navigation.Regions;
+using Uno.Extensions.Navigation.UI;
 
 namespace Uno.Extensions.Navigation;
 
 public static class FrameworkElementExtensions
 {
+	public static void AttachServices(this FrameworkElement element, IServiceProvider services)
```

Comment:
Why have both AttachServices and SetServiceProvider?

Reply:
Currently we do - thinking that we can remove either one. I felt AttachServices was more descriptive, whilst SetServiceProvider was the default setter for attached properties. Happy to remove whichever one we don't want to expose